Gems/Atom/RPI/Code/Source/RPI.Public/Scene.cpp

```diff
@@ -435,16 +435,14 @@ namespace AZ
             }
         }
 
-        void Scene::WaitTGEvent(AZ::TaskGraphEvent& completionTGEvent, AZStd::atomic_bool* workToWaitOn )
+        void Scene::WaitAndCleanTGEvent(AZ::TaskGraphEvent*& completionTGEvent)
```

Comment:
Should we use a `unique_ptr` for the completionTGEvent, accept it as a `std::unique_ptr<AZ::TaskGraphEvent>&&` and simply let it go out of scope here? This would annotate the lifetime a bit better.

Reply:
Compiler does not like :/
` error C2664: 'void AZ::RPI::Scene::WaitAndCleanTGEvent(std::unique_ptr<AZ::TaskGraphEvent,std::default_delete<T>> &&)': cannot convert argument 1 from 'std::unique_ptr<AZ::TaskGraphEvent,std::default_delete <T>>' to 'std::unique_ptr<AZ::TaskGraphEvent,std::default_delete<T>> &&'`

Reply:
Sounds like you're missing a `std::move`

Reply:
Updated to use the AZStd::unique_ptr